igannotator/rulesexecutor/rules.py

```diff
@@ -143,9 +146,38 @@ def apply(self, tree: LexicalTree, annotations: List[IGTag]):
 
         for c in tree.children:
             if c.relation in ["obj", "dobj", "obl"]:
-                annotations.append(
-                    IGTag(words=[(c.id, c.value)], tag_name=IGElement.OBJECT)
-                )
+                noun_type = self.nounClassifier(c.lemm)
+                if noun_type is not None:
+                    annotations.append(
+                        IGTag(
+                            words=[(c.id, c.value)],
+                            # words=[(cc.id, cc.value) for cc in c.get_all_descendants()],
+                            tag_name=noun_type
+                        )
+                    )
+
+    API_URL = "http://api.slowosiec.clarin-pl.eu:80/plwordnet-api/"
+    ACTOR_DOMAINS = ['os', 'grp']
+
+    def nounClassifier(self, word):
```

Comment:
I would move this method to another file (possible extension - hypernonyms)

Reply:
done